Deployment/Jenkins/master/parent-instance.groovy

```diff
@@ -0,0 +1,303 @@
+                      },
+                      {
+                        "ParameterKey": "ElbTemplate",
+                        "ParameterValue": "${env.ElbTemplate}"
+                      },
+                      {
+                        "ParameterKey": "IamRoleTemplate",
+                        "ParameterValue": "${env.IamRoleTemplate}"
+                      },
+                      {
+                        "ParameterKey": "SecurityGroupTemplate",
+                        "ParameterValue": "${env.SecurityGroupTemplate}"
+                      },
+                      {
+                        "ParameterKey": "TargetVPC",
+                        "ParameterValue": "${env.TargetVPC}"
+                      },
+                      {
+                      "ParameterKey": "BackupCronURL",
+                      "ParameterValue": "${env.BackupCronURL}"
+                      },
+                      {
+                      "ParameterKey": "SubnetId",
+                      "ParameterValue": "${env.SubnetId}"
+                      }
+                    ]
+                   /
+                }
+            }
+        stage ('Prepare AWS Environment') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash
+                        echo "Attempting to delete any active ${CfnStackRoot} stacks... "
+                        aws --region "${AwsRegion}" cloudformation delete-stack --stack-name "${CfnStackRoot}"
+
+                        aws cloudformation wait stack-delete-complete --stack-name ${CfnStackRoot} --region ${AwsRegion}
+                    '''
+                }
+            }
+        }
+        stage ('Launch Jenkins Master Parent Instance Stack') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash
+                        echo "Attempting to create stack ${CfnStackRoot}-${BUILD_NUMBER}..."
```

Comment:
In thinking about this, since our discussion, leaving these as _just_ `${CfnStackRoot}-${BUILD_NUMBER}` has the potential to result in sadness-inducing collisions once pipelines for other stack-elements are added to the pipeline-sets. Really probably want to do this as `${CfnStackRoot}-<RES_TYPE>-${BUILD_NUMBER}`